segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -205,12 +205,15 @@ public void readSegment(ReadSegment readSegment) {
     }
 
     private boolean verifyToken(String segment, long requestId, String delegationToken, String operation) {
-        if (!tokenVerifier.verifyToken(segment, delegationToken, READ)) {
-            log.warn(requestId, "Delegation token verification failed.");
-            handleException(requestId, segment, operation, new AuthenticationException("Token verification failed"));
-            return false;
+        boolean isTokenValid = false;
```

Comment:
nit: given that the isTokenValid is set to false only incase of `TokenException` we could as well make the default value of isTokenValid to true

Reply:
The idea here is to be restrictive by default, and be permissive only if the conditions for token validity are positively met. So, in this case, the token will be deemed invalid even if `tokenVerifier.verifyToken(...)` throws any other exception other than the one that is being caught, or `handleException(...)` itself throws an exception.

Reply:
There was an unnecessary statement to set the flag to false in the subsequent catch block, which I've now removed.